### PR TITLE
Update commands for code linting in contributing guidelines

### PR DESCRIPTION
**Pull request by Ben-Schneider-code**
# What does this PR do?
As I was looking to for the code linting commands for a pr, I was confused by what I think is missing linting commands. It looks like a bad copy/paste?

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by qgallouedec** [comment]
Good catch @Ben-Schneider-code, thanks for reporting. I'd like to take this opportunity to be a little more specific about what our precommit does. Here's a suggestion:

````md
TRL relies on `ruff` for maintaining consistent code formatting across its source files. Before submitting any PR, you should apply automatic style corrections and run code verification checks.

We provide a `precommit` target in the `Makefile` that simplifies this process by running all required checks and optimizations on only the files modified by your PR.

To apply these checks and corrections in one step, use:

```bash
$ make precommit
```

This command runs the following:
- Executes `pre-commit` hooks to automatically fix style issues with `ruff` and other tools.
- Runs additional scripts such as adding copyright information.

If you prefer to apply the style corrections separately or review them individually, the `pre-commit` hook will handle the formatting for the files in question.
````

**Comment by Ben-Schneider-code** [comment]
@qgallouedec  done 👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -179,19 +179,23 @@ Follow these steps to start contributing:
    ```bash
    $ make test
    ```
+<br>
```
Comment (qgallouedec) [inline CONTRIBUTING.md:182]:
```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -179,19 +179,23 @@ Follow these steps to start contributing:
...
 
-   TRL relies on `ruff` to format its source code
-   consistently. After you make changes, apply automatic style corrections and code verifications
-   that can't be automated in one go with:
+TRL relies on `ruff` for maintaining consistent code formatting across its source files. Before submitting any PR, you should apply automatic style corrections and run code verification checks.
```
Comment (qgallouedec) [inline CONTRIBUTING.md:184]:
```suggestion
    TRL relies on `ruff` for maintaining consistent code formatting across its source files. Before submitting any PR, you should apply automatic style corrections and run code verification checks.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -179,19 +179,23 @@ Follow these steps to start contributing:
...
 
-   This target is also optimized to only work with files modified by the PR you're working on.
+We provide a `precommit` target in the `Makefile` that simplifies this process by running all required checks and optimizations on only the files modified by your PR.
```
Comment (qgallouedec) [inline CONTRIBUTING.md:186]:
```suggestion
    We provide a `precommit` target in the `Makefile` that simplifies this process by running all required checks and optimizations on only the files modified by your PR.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -179,19 +179,23 @@ Follow these steps to start contributing:
...
 
-   If you prefer to run the checks one after the other, the following command apply the
-   style corrections:
+To apply these checks and corrections in one step, use:
```
Comment (qgallouedec) [inline CONTRIBUTING.md:188]:
```suggestion
    To apply these checks and corrections in one step, use:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -179,19 +179,23 @@ Follow these steps to start contributing:
...
-   ```
+```bash
+$ make precommit
+```
+
+This command runs the following:
+- Executes `pre-commit` hooks to automatically fix style issues with `ruff` and other tools.
+- Runs additional scripts such as adding copyright information.
+
+If you prefer to apply the style corrections separately or review them individually, the `pre-commit` hook will handle the formatting for the files in question.
```
Comment (qgallouedec) [inline CONTRIBUTING.md:198]:
````suggestion
    This command runs the following:
    - Executes `pre-commit` hooks to automatically fix style issues with `ruff` and other tools.
    - Runs additional scripts such as adding copyright information.

    If you prefer to apply the style corrections separately or review them individually, the `pre-commit` hook will handle the formatting for the files in question.
````

**Review by qgallouedec — CHANGES_REQUESTED** [review]

**Review by qgallouedec — APPROVED** [review]

**Comment by qgallouedec** [comment]
Thanks a lot @Ben-Schneider-code!

**Pull request merged**